ssvsigner/server.go

```diff
@@ -85,35 +90,70 @@ func (r *Server) Handler() func(ctx *fasthttp.RequestCtx) {
 				time.Since(start),
 			)
 		}()
-		r.router.Handler(ctx)
+		s.router.Handler(ctx)
+	}
+}
+
+// ListenAndServe starts the server on the specified address.
+// If TLS is configured, it will use HTTPS.
+func (s *Server) ListenAndServe(addr string) error {
+	handler := s.Handler()
+
+	if s.tlsConfig != nil {
+		s.logger.Info("starting with TLS", zap.String("addr", addr))
+		ln, err := net.Listen("tcp", addr)
+		if err != nil {
+			return err
+		}
+
+		tlsLn := tls.NewListener(ln, s.tlsConfig)
+		return fasthttp.Serve(tlsLn, handler)
 	}
+
+	s.logger.Info("starting without TLS", zap.String("addr", addr))
+	return fasthttp.ListenAndServe(addr, handler)
+}
+
+// SetTLS configures the server to use TLS by loading the given certificate, key, and CA paths into the TLS configuration.
+func (s *Server) SetTLS(certPath, keyPath, caPath string) error {
+	if certPath == "" && keyPath == "" {
+		return nil
+	}
```

Comment:
we should return error in the case that one of the flags is missing and not `nil`. the check should be before going into this function. Also it should be OR and not AND.